src/parallel/once_cell.rs

```diff
@@ -0,0 +1,47 @@
+use std::{
+    cell::UnsafeCell,
+    marker::PhantomData,
+    mem::MaybeUninit,
+    panic::{RefUnwindSafe, UnwindSafe},
+    sync::Once,
+};
+
+pub(crate) struct OnceCell<T> {
+    once: Once,
+    value: UnsafeCell<MaybeUninit<T>>,
+    _marker: PhantomData<T>,
+}
+
+impl<T> OnceCell<T> {
+    pub(crate) const fn new() -> Self {
+        Self {
+            once: Once::new(),
+            value: UnsafeCell::new(MaybeUninit::uninit()),
+            _marker: PhantomData,
+        }
+    }
+
+    pub(crate) fn get_or_init(&self, f: impl FnOnce() -> T) -> &T {
+        self.once.call_once_force(|_| {
+            unsafe { &mut *self.value.get() }.write(f());
```

Comment:
In the unlikely event the Once is poisoned, I don't think we should try to continue, as cc-rs generally is used for run-once invocations. If we're vendoring everything we should just mimic the `get_or_try_init` API.

Reply:
Thanks, updated